TriblerGUI/widgets/trustpage/js/legend.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Draw the legend based on the current visible network.
+ * Handle clicks to show and hide the legend.
+ */
+
+/**
+ * Draw the legend based on the quantitative data of the graph currently visible
+ * @param data the data of the graph
+ */
+function drawLegend (data) {
+    var svg = d3.select("#legend");
+
+    // Link size
+
+    var linkSize = d3.scaleLinear()
+        .domain([data.min_transmission, data.max_transmission])
+        .range([config.link.strokeWidthMin, config.link.strokeWidthMax]);
+
+    svg.append("g")
+        .attr("class", "legendSizeLink")
+        .attr("fill", config.legend.textColor)
+        .attr("transform", "translate(10, 20)");
+
+    var legendSizeLink = d3.legendSize()
+        .classPrefix("linkSize")
+        .title("Data over connection")
+        .scale(linkSize)
+        .shape("line")
+        .labelWrap(70)
+        .shapeWidth(70)
+        .shapePadding(15)
+        .labelOffset(20)
+        .labels(function(i) {
+            return formatBytes(i.generatedLabels[i.i], 3);
+        });
+
+    applyLegendStyle(legendSizeLink);
+
+    svg.select(".legendSizeLink")
+        .call(legendSizeLink);
+
+    // Data transmission nodes
+
+    var nodeSize = d3.scaleLinear()
+        .range([config.node.circle.minRadius, config.node.circle.maxRadius]);
+
+    var traffic_diff = data.traffic_max - data.traffic_min;
+
+    svg.append("g")
+        .attr("class", "legendSizeNode")
+        .attr("transform", "translate(10, 105)");
+
+    console.log("Node size:" + nodeSize);
+
+    var legendSizeNode = d3.legendSize()
+        .classPrefix("nodeSize")
+        .title("Data transmitted")
+        .scale(nodeSize)
+        .shape("circle")
+        .labelWrap(70)
+        .shapePadding(43)
+        .labelOffset(20)
+        .labels(function(i) { return formatBytes(data.traffic_min + i.i * traffic_diff, 3); });
+
+    applyLegendStyle(legendSizeNode);
+
+    svg.select(".legendSizeNode")
+        .call(legendSizeNode);
+
+    // Trust colors
+
+    var nodeColorScale = d3.scaleLinear()
+        .domain(config.node.color.domain)
+        .range(config.node.color.range);
+
+    range = [];
+    for (var i = 0; i < 6; i++) {
```

Comment:
This for-loop does not go to 1 on the scale, but the "Trusted" node is instead 5/6th on the scale. Is this done on purpose?